numba/cuda/tests/cudapy/test_dispatcher.py

```diff
@@ -92,20 +100,54 @@ def f(x, y):
 
         self.assertEqual(len(f.specializations), 0)
 
-        # 'A' order specialization
-        f_f32a_f32a = f.specialize(float32[:], float32[:])
+        # 'F' order specialization
+        f_arg = np.zeros((2, 2), order='F')
+        f_f32a_f32a = f.specialize(f_arg, f_arg)
         self.assertEqual(len(f.specializations), 1)
 
         # 'C' order specialization
-        f_f32c_f32c = f.specialize(float32[::1], float32[::1])
+        c_arg = np.zeros((2, 2), order='C')
+        f_f32c_f32c = f.specialize(c_arg, c_arg)
         self.assertEqual(len(f.specializations), 2)
         self.assertIsNot(f_f32a_f32a, f_f32c_f32c)
 
         # Reuse 'C' order specialization
-        f_f32c_f32c_2 = f.specialize(float32[::1], float32[::1])
+        f_f32c_f32c_2 = f.specialize(c_arg, c_arg)
         self.assertEqual(len(f.specializations), 2)
         self.assertIs(f_f32c_f32c, f_f32c_f32c_2)
 
+    def test_specialize_adds_overload(self):
+        # Ensure that specializing a dispatcher adds the overload generated by
+        # specializing the dispatcher to the generic dispatcher, so we don't
+        # need to generate the overload on multiple uses.
+        @cuda.jit
+        def f(x, y):
+            pass
+
+        self.assertEqual(len(f.overloads), 0)
+
+        # 'F' order specialization
+        f_arg = np.zeros((2, 2), order='F')
+        f_arg_ty = typeof(f_arg)
+        f_f32a_f32a = f.specialize(f_arg, f_arg)
+        self.assertEqual(len(f.overloads), 1)
+        self.assertIs(f_f32a_f32a.overloads[f_arg_ty, f_arg_ty],
```

Comment:
```suggestion
        f_f32f_f32f = f.specialize(f_arg, f_arg)
        self.assertEqual(len(f.overloads), 1)
        self.assertIs(f_f32f_f32f.overloads[f_arg_ty, f_arg_ty],
```
suggest naming scheme consistency with `F`-ordered input.